### PR TITLE
[Doc Revamp][webhooks][1684078]

### PR DESCRIPTION
**Pull request by Mahesh2516**
doc revamp

### CODE REVIEW

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 1]
```diff
@@ -1,55 +1,77 @@
...
+ Incoming webhooks are simple way for an external app to share messages in Teams. These webhooks are often used as tracking and notification tools. Incoming webhooks provide a unique URL to which you send a JSON payload with the message that you want to POST, typically in a card format. Cards are user interface (UI) containers that include content and actions related to a single topic and are a way to present message data in a consistent way. Teams uses cards within the following three capabilities:
 
 * Bots
 * Messaging extensions
 * Connectors
 
-## Incoming webhook key features
+## Key features of webhooks
+
+The following table provides feature and description of webhooks:
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:25]: Do we need this sentence as the title itself is self explanatory?

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 2]
```diff
@@ -1,55 +1,77 @@
...
+|Scoped configuration|Incoming webhooks are scoped and configured at the channel level. Outgoing webhooks are scoped and configured at the team level.|
+|Secure resource definitions|Messages are formatted as JSON payloads. This declarative messaging structure prevents insertion of malicious code as there is no code execution on the client.|
+|Actionable messaging support|If you choose to send messages through cards, you must use the actionable message card format. Actionable message cards are supported in all Office 365 groups including Teams. Here are links to the [legacy actionable message card reference](/outlook/actionable-messages/message-card-reference) and the [message card playground](https://messagecardplayground.azurewebsites.net).|
+|Independent HTTPS messaging support| Cards are a great way to present information in a clear and consistent way. Any tool or framework that can send HTTPS POST requests can send messages to Teams through an incoming webhook.|
+|Markdown support|All text fields in actionable messaging cards support basic Markdown. Do not use HTML markup in your cards. HTML is ignored and treated as plain text.|
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:33]: Please follow alphabetical order and maintain parallelism in the description.

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 3]
```diff
@@ -1,55 +1,77 @@
...
 
-> [!Note]
+> [!NOTE]
 > Teams bots, messaging extensions, incoming webhooks, and the Bot Framework support Adaptive Cards, an open cross-card platform framework. [Teams connectors](../../webhooks-and-connectors/how-to/connectors-creating.md) do not currently support Adaptive Cards. However, it is possible to create a [flow](https://flow.microsoft.com/blog/microsoft-flow-in-microsoft-teams/) that posts Adaptive Cards to a Teams channel.
 
-## Add an incoming webhook to a Teams channel
+> [!VIDEO https://www.youtube-nocookie.com/embed/y5pbJI43Zvg&t=204s]
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:38]: There is some problem with the video. Please resolve. What si this video about? Does it highlight the features?

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 4]
```diff
@@ -1,55 +1,77 @@
...
+
+## Add an incoming webhook
+
+> [!IMPORTANT]
+> If your team's **Settings** > **Member permissions** > **Allow members to create, update, and remove connectors** is selected, any team member can add, modify, or delete a connector.
+
+Perform the following steps to add an incoming webhook to a Teams channel.
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:45]: Follow the style guide for stem sentences.

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 5]
```diff
@@ -1,55 +1,77 @@
...
 1. Choose **Connectors** from the drop-down menu.
 1. On the left, under **Manage**, choose **Configured**.
-1. Select the *number Configured* to see a list of your current connectors.
+1. Select the **number Configured** to see a list of your current connectors.
 1. Select **Manage** next to the connector that you want to delete.
-1. Select the **Remove** button and you will be presented with a *Remove Configuration* dialog box.
-1. Optionally, complete the dialog box fields and checkboxes prior to selecting the **Remove** button. The webhook will be deleted from the team channel.
+1. Select the **Remove** button. The **Remove Configuration** dialog box appears. 
+1. Optionally, complete the dialog box fields and checkboxes prior to selecting the **Remove** button.
+1. The webhook is deleted from the team channel.
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:74]: Style guide is not followed and add images where applicable.

Are these steps tested?

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 6]
```diff
@@ -1,55 +1,77 @@
...
 
 ## Distribution
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:76]: Modify the Distribution section. 
Shouldn't the heading be Distribute an incoming webhook? And why do we need to distribute?

How do we use these three options to distribute. Where is that mentioned?

Shouldn't the second link point to https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-creating#publish-connectors-for-your-organization? If yes, then change it to Distribute/Publish.

File: msteams-platform/webhooks-and-connectors/how-to/add-outgoing-webhook.md [hunk 7]
```diff
@@ -1,31 +1,38 @@
...
+title: Create an outgoing webhook
+description: describes how to create an outgoing webhook
 ms.topic: conceptual
-ms.author: lajanuar
 localization_priority: Normal
+ms.author: lajanuar
 keywords: teams tabs outgoing webhook actionable message verify webhook
+
 ---
-# Add custom bots to Teams with outgoing webhooks
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/add-outgoing-webhook.md:11]: Global: Self review is not performed for this file. Please revisit and apply the comments given in the previous file.

File: msteams-platform/webhooks-and-connectors/how-to/connectors-using.md [hunk 8]
```diff
@@ -1,19 +1,21 @@
...
-title: Sending messages to Connectors and Webhooks
+title: Send messages to Connectors and Webhooks
 description: Describes how to use Office 365 Connectors in Microsoft Teams
 ms.topic: how-to
 localization_priority: Normal
 keywords: teams o365 connector
 ---
 
-# Sending messages to connectors and webhooks
+# Send messages to connectors and webhooks
```
Comment (surbhigupta12) [inline msteams-platform/webhooks-and-connectors/how-to/connectors-using.md:9]: Self review is not done. All the images in this module are very old. get the latest images.

**Review by surbhigupta12 — CHANGES_REQUESTED** [review]
Please revisit this module.

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md (hunk 3 above)
Comment (Mahesh2516) [reply msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md]: This video is removed.

File: msteams-platform/webhooks-and-connectors/how-to/create-actionable-messages.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+# [Verify HMAC token](#tab/verifyhmactoken)
+**Create a method to verify the outgoing webhook HMAC token**
+
+#### HMAC signature for testing with code example
+
+Using example of inbound message and ID: "contoso" of SigningKeyDictionary of {"contoso", "vqF0En+Z0ucuRTM/01o2GuhMH3hKKk/N2bOmlM31zaA=" }.
+
+Use the value "HMAC 03TCao0i55H1eVKUusZOTZRjtvYTs+mO41mPL+R1e1U=" in the authorization of request header.
+
+To ensure that your service is receiving calls only from actual Teams clients, Teams provides an HMAC code in the HTTP `hmac` header. Always include the code in your authentication protocol.
```
Comment (alexanderdavide) [inline msteams-platform/webhooks-and-connectors/how-to/create-actionable-messages.md:71]:
```suggestion
To ensure that your service is receiving calls only from actual Teams clients, Teams provides an HMAC code in the HTTP `authorization` header. Always include the code in your authentication protocol.
```

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 10]
```diff
@@ -1,68 +1,91 @@
...
+|Independent HTTPS messaging support|Cards provide information clearly and consistently. Any tool or framework that can send HTTPS POST requests, can send messages to Teams through an incoming webhook.|
+|Markdown support|All text fields in actionable messaging cards support basic Markdown. Do not use HTML markup in your cards. HTML is ignored and treated as plain text.|
+|Scoped configuration|Incoming webhook is scoped and configured at the channel level.|
+|Secure resource definitions|Messages are formatted as JSON payloads. This declarative messaging structure prevents the insertion of malicious code.|
+
+> [!NOTE]
+> Teams bots, messaging extensions, incoming webhook, and the Bot Framework support Adaptive Cards, an open cross card platform framework. Currently, [Teams connectors](../../webhooks-and-connectors/how-to/connectors-creating.md) do not support Adaptive Cards. However, it is possible to create a [flow](https://flow.microsoft.com/blog/microsoft-flow-in-microsoft-teams/) that posts Adaptive Cards to a Teams channel.
```
Comment (pradeepananth) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:32]: nit: camel case Incoming Webhook

Comment (pradeepananth) [inline msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:32]: Add this link for reference to Adaptive cards and Incoming webhooks:  https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/what-are-cards#adaptive-cards-and-incoming-webhooks

File: msteams-platform/webhooks-and-connectors/what-are-webhooks-and-connectors.md [hunk 11]
```diff
@@ -1,35 +1,48 @@
...
 
-Outgoing webhooks allow your users to send text messages from a channel to your web services. Once configured, your users will be able to @mention your outgoing webhook and send a message to your service. Your service will have five seconds to send a response to the message, potentially containing text or a card.
+## Outgoing webhooks
 
-Outgoing webhooks are configured on a per-team basis, cannot be included as part of a normal Teams app. They are best suited for completing team-specific workloads that don't require large amounts of information to be collected or exchanged.
+Webhooks help Teams to integrate with external apps. With outgoing webhooks, you can send text messages from a channel to the web services. After configuring the outgoing webhooks, users can @mention outgoing webhook and send a message to web services. The service responds within five seconds to the message with a text or a card.
```
Comment (pradeepananth) [inline msteams-platform/webhooks-and-connectors/what-are-webhooks-and-connectors.md:16]: "five seconds" has been updated to "ten seconds", please update the same.

File: msteams-platform/webhooks-and-connectors/what-are-webhooks-and-connectors.md [hunk 12]
```diff
@@ -1,35 +1,48 @@
...
 
-For more information, see [Create an outgoing webhook](~/webhooks-and-connectors/how-to/add-outgoing-webhook.md).
+> [!NOTE]
+> Outgoing webhooks are configured on a per team basis and cannot be included as part of a normal Teams app.
 
 ## Connectors
 
-Connectors allow users to subscribe to receive notifications and messages from your web services. They expose an HTTPS endpoint for your service to post messages to - typically in the form of cards.
+Connectors allow users to subscribe to receive notifications and messages from the web services. They expose the HTTPS endpoint for the service to post messages to the users, typically in the form of cards.
```
Comment (pradeepananth) [inline msteams-platform/webhooks-and-connectors/what-are-webhooks-and-connectors.md:23]: post messages to Teams Channels

File: msteams-platform/webhooks-and-connectors/what-are-webhooks-and-connectors.md [hunk 13]
```diff
@@ -1,35 +1,48 @@
...
 
 ### Incoming webhooks
 
-Incoming webhooks are the simplest type of connector. For any channel in team (if they are enabled for that team) you can choose to expose an HTTPS endpoint that will accept correctly formatted JSON and insert messages into that channel. They are a quick and easy way to connect a channel to your service, and are best used for scenarios that are unique to a particular team. For example, you could create an incoming webhook in your DevOps channel and configure your build, deployment and monitoring services to send alerts.
-
-For more information, see [Create an incoming webhook](~/webhooks-and-connectors/how-to/add-incoming-webhook.md).
+Incoming webhooks help in posting messages from apps to Teams. If incoming webhooks are enabled for a team in any channel, it exposes the HTTPS endpoint, which accepts correctly formatted JSON and inserts the messages into that channel. You can create an incoming webhook in your DevOps channel, configure your build, and simultaneously deploy and monitor services to send alerts.
```
Comment (pradeepananth) [inline msteams-platform/webhooks-and-connectors/what-are-webhooks-and-connectors.md:27]: nit: For example, you can create Incoming Webhook in your ....

**Review by pradeepananth — CHANGES_REQUESTED** [review]

**Review by pradeepananth — DISMISSED** [review]
nit: use camel case for incoming wehbooks: Incoming Webhooks

File: msteams-platform/webhooks-and-connectors/how-to/connectors-using.md [hunk 14]
```diff
@@ -160,82 +160,43 @@ The following steps use [cURL](https://curl.haxx.se/). We assume that you have t
...
+    > [!NOTE]
+    > If the POST succeeds, you must see a simple **1** output by `Invoke-RestMethod`.
 
-- [Include two icons](../../concepts/build-and-test/apps-package.md#app-icons).
-- Modify the `icons` portion of the manifest to refer to the file names of the icons instead of URLs.
+1. Check the Microsoft Teams channel associated with the webhook URL. You can see the new card posted to the channel. Before you use the connector to test or publish your app, you must do the following:
 
-The following manifest.json file contains the basic elements needed to test and submit your app:
-
-> [!NOTE]
```
Comment (Wajeed-msft) [inline msteams-platform/webhooks-and-connectors/how-to/connectors-using.md:184]: I think this entire section is misplace. Could you please check and remove this? This is already present here: https://review.docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-creating?branch=pr-en-us-3657#including-the-connector-in-your-manifest

File: msteams-platform/webhooks-and-connectors/how-to/connectors-using.md [hunk 15]
```diff
@@ -98,55 +136,17 @@ You can also use this JSON to create cards containing rich inputs, such as text
...
-> 
-> For the HttpPOST action, the bearer token is included with the requests. This token includes the Azure AD identity of the Office 365 user who took the action.
-
-## Setting up a custom incoming webhook
-
-Follow these steps to see how to send a simple card to a Connector:
-
-1. In Microsoft Teams, choose **More options** (**&#8943;**) next to the channel name and then choose **Connectors**.
-1. Scroll through the list of Connectors to **Incoming Webhook**, and choose **Add**.
-1. Enter a name for the webhook, upload an image to associate with data from the webhook, and choose **Create**.
```
Comment (Wajeed-msft) [inline msteams-platform/webhooks-and-connectors/how-to/connectors-using.md:143]: This seems repeated from Create incoming webhook page. Ref: https://review.docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?branch=pr-en-us-3657#add-an-incoming-webhook-to-a-teams-channel

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md [hunk 16]
```diff
@@ -1,68 +1,92 @@
...
-|Independent HTTPS messaging support| Cards are a great way to present information in a clear and consistent way. Any tool or framework that can send HTTPS POST requests can send messages to Teams via an incoming webhook.|
-|Markdown support|All text fields in actionable messaging cards support basic Markdown. **Don't use HTML markup in your cards**. HTML is ignored and treated as plain text.|
+|Actionable messaging support|Actionable message cards are supported in all Office 365 groups including Teams. If you send messages through cards, you must use the actionable message card format. For more information, see [legacy actionable message card reference](/outlook/actionable-messages/message-card-reference) and [message card playground](https://messagecardplayground.azurewebsites.net).|
```
[the remark this answers is not in the document]
Comment (v-abirade) [reply msteams-platform/webhooks-and-connectors/how-to/add-incoming-webhook.md:26]: Added table details

**Review by Wajeed-msft — APPROVED** [review]

**Review by v-abirade — APPROVED** [review]
Approve

**Review by surbhigupta12 — APPROVED** [review]
Approved

**Pull request merged**